File: create_user.py
#!/usr/bin/env python3
"""
Script para criar um usuário no banco de dados SynapScale
"""
import os
import sys
import uuid
import sqlite3
from datetime import datetime, timezone

# Importar apenas as funções de hash necessárias
sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'src'))

from passlib.context import CryptContext

# Configuração do contexto de criptografia
pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

# Configuração do banco de dados
DATABASE_URL = "sqlite:///synapse.db"

def get_password_hash(password: str) -> str:
    """Gera hash da senha para armazenamento seguro."""
    return pwd_context.hash(password)


def create_user(email: str, password: str, first_name: str = None, last_name: str = None):
    """Cria um novo usuário no banco de dados SQLite diretamente via SQL."""
    conn = sqlite3.connect("synapse.db")
    cursor = conn.cursor()
    try:
        # Verificar se o usuário já existe
        cursor.execute("SELECT id FROM users WHERE email = ?", (email,))
        if cursor.fetchone():
            print(f"❌ Erro: Usuário com email '{email}' já existe!")
            return False
        # Gerar hash da senha
        password_hash = get_password_hash(password)
        # Gerar UUID
        user_id = str(uuid.uuid4())
        now = datetime.now(timezone.utc).isoformat()
        # Inserir usuário
        cursor.execute(
            """
            INSERT INTO users (
                id, email, password_hash, first_name, last_name, is_active, is_verified, role, subscription_plan, preferences, created_at, updated_at
            ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
            """,
            (
                user_id,
                email,
                password_hash,
                first_name,
                last_name,
                1,  # is_active
                1,  # is_verified
                "user",
                "free",
                "{}",  # preferences vazio
                now,
                now
            )
        )
        conn.commit()
        print("✅ Usuário criado com sucesso!")
        print(f"   ID: {user_id}")
        print(f"   Email: {email}")
        print(f"   Nome: {first_name or ''} {last_name or ''}")
        print("   Ativo: True")
        print("   Verificado: True")
        print("   Role: user")
        print(f"   Data de criação: {now}")
        return True
    except Exception as e:
        print(f"❌ Erro ao criar usuário: {str(e)}")
        conn.rollback()
        return False
    finally:
        conn.close()

def main():
    """Função principal"""
    print("🚀 Criando usuário no SynapScale Backend")
    print("=" * 50)
    
    # Dados do usuário
    email = "joaovictor@liderimobiliaria.com.br"
    password = "@Jvcm1811"
    first_name = "João Victor"
    last_name = "Lider Imobiliária"
    
    print(f"📧 Email: {email}")
    print(f"👤 Nome: {first_name} {last_name}")
    print(f"🔐 Senha: {'*' * len(password)}")
    print()
    
    # Criar usuário
    success = create_user(
        email=email,
        password=password,
        first_name=first_name,
        last_name=last_name
    )
    
    if success:
        print()
        print("🎉 Usuário criado e pronto para usar!")
        print("   Você pode fazer login com esses dados no sistema.")
    else:
        print()
        print("❌ Falha ao criar usuário. Verifique os logs acima.")

if __name__ == "__main__":
    main()

File: hello-prisma/create-user.js
const { PrismaClient } = require('@prisma/client');
const bcrypt = require('bcryptjs');

const prisma = new PrismaClient();

async function createUser() {
  try {
    console.log('🚀 Criando usuário no PostgreSQL...\n');
    
    // Conectar ao banco
    await prisma.$connect();
    console.log('✅ Conectado ao PostgreSQL');
    
    // Email e senha fornecidos
    const email = 'joaovictor@liderimobiliaria.com.br';
    const password = '@Jvcm1811';
    
    // Verificar se o usuário já existe
    const existingUser = await prisma.users.findFirst({
      where: { email: email }
    });
    
    if (existingUser) {
      console.log('⚠️  Usuário já existe! Atualizando senha...');
      
      // Criptografar nova senha
      const hashedPassword = await bcrypt.hash(password, 12);
      
      // Atualizar usuário existente
      const updatedUser = await prisma.users.update({
        where: { id: existingUser.id },
        data: {
          password_hash: hashedPassword,
          is_active: true,
          is_verified: true,
          updated_at: new Date()
        }
      });
      
      console.log('✅ Usuário atualizado com sucesso!');
      console.log(`   ID: ${updatedUser.id}`);
      console.log(`   Email: ${updatedUser.email}`);
      console.log(`   Status: ${updatedUser.is_active ? 'Ativo' : 'Inativo'}`);
      console.log(`   Verificado: ${updatedUser.is_verified ? 'Sim' : 'Não'}`);
      
    } else {
      console.log('📝 Criando novo usuário...');
      
      // Criptografar senha
      const hashedPassword = await bcrypt.hash(password, 12);
      
      // Criar novo usuário
      const newUser = await prisma.users.create({
        data: {
          id: 'user_' + Date.now(),
          email: email,
          password_hash: hashedPassword,
          first_name: 'João Victor',
          last_name: 'Lider Imobiliária',
          is_active: true,
          is_verified: true,
          role: 'admin',
          subscription_plan: 'premium',
          created_at: new Date(),
          updated_at: new Date()
        }
      });
      
      console.log('✅ Usuário criado com sucesso!');
      console.log(`   ID: ${newUser.id}`);
      console.log(`   Email: ${newUser.email}`);
      console.log(`   Nome: ${newUser.first_name} ${newUser.last_name}`);
      console.log(`   Role: ${newUser.role}`);
      console.log(`   Plano: ${newUser.subscription_plan}`);
      console.log(`   Status: ${newUser.is_active ? 'Ativo' : 'Inativo'}`);
      console.log(`   Verificado: ${newUser.is_verified ? 'Sim' : 'Não'}`);
    }
    
    console.log('\n🎉 Processo concluído com sucesso!');
    console.log('🔐 Senha criptografada e armazenada com segurança');
    
  } catch (error) {
    console.error('❌ Erro ao criar usuário:', error);
  } finally {
    await prisma.$disconnect();
  }
}

createUser();
